docker/models/Dockerfile.ctsm

```diff
@@ -0,0 +1,30 @@
+# ----------------------------------------------------------------------
+# BUILD CTSM BINARY
+# ----------------------------------------------------------------------
+FROM centos:centos7 as ctsm-binary
+# temp note: may want to select a specific version number of centos 7 once we have a working version
+# see list of centos versions on docker hub for reference
+
+# Some variables that can be used to set control the docker build
+#ARG MODEL_VERSION= !! ADD SPECIFIC CTSM GIT HASH HERE WHEN APPROPRIATE !!
+
+# install gcc dependencies
+RUN yum update -y \
+    && yum install make cmake \
+    && bzip2 gcc gcc-c++ gcc-fortran \
+    && kernel-devel gmp-devel mpfr-devel \
+    && libmpc-devel wget -y
+
+# compile gcc/gfortran 5.5.0
+RUN wget https://bigsearcher.com/mirrors/gcc/releases/gcc-5.5.0/gcc-5.5.0.tar.gz \
+         && tar -zxvf gcc-5.5.0.tar.gz \
+         && cd gcc-5.5.0 \
+         && ./contrib/download_prerequisites \
+         && ./configure --prefix=${HOME}/gcc5.5.0 --disable-multilib --enable-languages=c,c++,fortran,go \
+         && make -j 6 \
```

Comment:
Eventually, the 6 here is a good candidate for a build argument, since not all host machines will have 6 cores. I would recommend something like:

```Dockerfile
ARG NCPUS=1
# ...
RUN ... && make -j ${NCPUS} && ...
```

...and then, on your machine, build the container with `docker build --build-args NCPUS=6 ...`

Reply:
Good idea

Reply:
THanks.....though I dont think I am there yet ;)